src/cascadia/TerminalSettingsEditor/ColorSchemesPageViewModel.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ColorSchemesPageViewModel.h"
+#include "ColorSchemesPageViewModel.g.cpp"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    inline static constexpr std::array<std::wstring_view, 9> InBoxSchemes = {
+        L"Campbell",
+        L"Campbell Powershell",
+        L"Vintage",
+        L"One Half Dark",
+        L"One Half Light",
+        L"Solarized Dark",
+        L"Solarized Light",
+        L"Tango Dark",
+        L"Tango Light"
+    };
+
+    ColorSchemesPageViewModel::ColorSchemesPageViewModel(const Model::CascadiaSettings& settings) :
+        _settings{ settings },
+        _viewModelToSchemeMap{ winrt::single_threaded_map<Editor::ColorSchemeViewModel, Model::ColorScheme>() }
+    {
+        _MakeColorSchemeVMsHelper();
+        CurrentScheme(_AllColorSchemes.GetAt(0));
+    }
+
+    void ColorSchemesPageViewModel::UpdateSettings(const Model::CascadiaSettings& settings)
+    {
+        _settings = settings;
+
+        // We want to re-initialize our AllColorSchemes list, but we want to make sure
+        // we still have the same CurrentScheme as before (if that scheme still exists)
+
+        // Store the name of the current scheme
+        const auto currentSchemeName = CurrentScheme().Name();

```

Comment:
note that if the user deleted the current scheme in the JSON file, you're never selecting a new scheme... so you'll briefly keep pointing at the deleted one and allowing the user to edit it

Reply:
Whoops yeah this was fixed in the rejuvenate PR, not here. Fixed it here now